File: BEmuJava/bemu/src/main/java/com/bloomberglp/blpapi/IntradayBarRequestElementInt.java
//------------------------------------------------------------------------------
// <copyright project="BEmu_maven" file="/BEmu_maven/bemu/src/main/java/com/bloomberglp/blpapi/IntradayBarRequestElementInt.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bloomberglp.blpapi;

import com.bloomberglp.blpapi.Schema;

public class IntradayBarRequestElementInt extends IntradayBarRequestElementString
{
    private final int _value;

    IntradayBarRequestElementInt(String elementName, int value)
    {
    	super(elementName, String.valueOf(value));
        this._value = value;
    }

    //I can't override GetElementAsInt32 here because the Bloomberg Request object stores ints as strings, not ints.  You can't convert the string to an int
    protected int getInt()
    {
    	return this._value;
    }

    public Schema.Datatype datatype()
    {
    	return Schema.Datatype.INT32;
    }
}

File: BEmu_cpp/src/IntradayBarRequest/IntradayBarRequestElementInt.cpp
//------------------------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/IntradayBarRequest/IntradayBarRequestElementInt.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------------------------

#include "BloombergTypes/Name.h"
#include "BloombergTypes/ElementPtr.h"
#include "IntradayBarRequest/IntradayBarRequestElementInt.h"

namespace BEmu
{
	namespace IntradayBarRequest
	{
		IntradayBarRequestElementInt::IntradayBarRequestElementInt(const std::string& elementName, int ivalue) : CanConvertToStringType(ivalue, false)
		{
			this->_elementName = elementName;
			this->_ivalue = ivalue;
		}

		IntradayBarRequestElementInt::~IntradayBarRequestElementInt()
		{
		}

		int IntradayBarRequestElementInt::getInt() const
		{
			return this->_ivalue;
		}

		Name IntradayBarRequestElementInt::name() const
		{
			Name result(this->_elementName.c_str());
			return result;
		}

		size_t IntradayBarRequestElementInt::numValues() const
		{
			return 1;
		}

		size_t IntradayBarRequestElementInt::numElements() const
		{
			return 0;
		}

		bool IntradayBarRequestElementInt::isArray() const
		{
			return false;
		}

		bool IntradayBarRequestElementInt::isComplexType() const
		{
			return false;
		}

		const char * IntradayBarRequestElementInt::getValueAsString(int index) const
		{
			return this->ValueAsString().c_str();
		}

		std::ostream& IntradayBarRequestElementInt::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			this->prettyPrintHelper(stream, level, spacesPerLevel, this->ValueAsString());
			return stream;
		}

	}
}